tiled/server/authentication.py

```diff
@@ -894,8 +894,29 @@ async def refresh_session(
     return new_tokens
 
 
-@base_authentication_router.delete("/session/revoke/{session_id}")
+@base_authentication_router.post("/session/revoke")
 async def revoke_session(
+    request: Request,
+    refresh_token: schemas.RefreshToken,
+    settings: BaseSettings = Depends(get_settings),
+    db=Depends(get_database_session),
+):
+    "Mark a Session as revoked so it cannot be refreshed again."
+    request.state.endpoint = "auth"
+    payload = decode_token(refresh_token.refresh_token, settings.secret_keys)
+    session_id = payload["sid"]
+    # Find this session in the database.
+    session = await lookup_valid_session(db, session_id)
+    if session is None:
+        raise HTTPException(409, detail=f"No session {session_id}")
+    session.revoked = True
+    db.add(session)
```

Comment:
In SQLAlchemy, is [`Session.add()`](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.add) an "upsert" operation -- it will perform INSERT or UPDATE as needed?

Reply:
https://docs.sqlalchemy.org/en/20/orm/session_basics.html#adding-new-or-existing-items

Reply:
(As usual with SQLAlchemy, the documentation is sprawling and specific things can be hard to find.)